### PR TITLE
Use Travis auto deploy command to generate a phar on each build, and …

### PR DESCRIPTION
**Pull request by wysow**
…configure Travis to upload it if a release is done.

### CODE REVIEW

**Comment by wysow** [comment]
@mnapoli what do you think about this?

**Comment by wysow** [comment]
@mnapoli We could think to use https://github.com/millesime/millesime/pull/1 when merged to build an optimized phar file maybe...

File: src/Application/Cli/TravisAutoDeployCommand.php [hunk 1]
```diff
@@ -58,13 +58,6 @@ protected function configure()
                 getcwd()
             )
             ->addOption(
-                'php-version',
-                null,
-                InputOption::VALUE_REQUIRED,
-                'Specify for which php version you want to deploy documentation, mainly to avoid multiple deploys',
-                '5.4'
-            )
```
Comment (mnapoli) [inline src/Application/Cli/TravisAutoDeployCommand.php]: Why remove this option? Even if we don't use it I imagine it can be used by some users?

File: src/Application/Cli/TravisAutoDeployCommand.php [hunk 2]
```diff
@@ -106,20 +99,17 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
 
         $output->writeln('');
 
         // Deploy it
         $this->deployer->deploy($repository, $output, $repositoryUrl, $targetBranch);
+
+        // generate couscous.phar to be uploaded when a new release is done
+        $this->commandRunner->run('bin/compile');
+        $output->writeln('<comment>PHAR file generated</comment>');
+
```
Comment (mnapoli) [inline src/Application/Cli/TravisAutoDeployCommand.php]: I'm not sure I understand this -> this will be executed for users of Couscous, that script will not exist

**Comment by mnapoli** [comment]
> @mnapoli We could think to use millesime/millesime#1 when merged to build an optimized phar file maybe...

I think it's not related to this PR? Let's discuss that in another PR it will be easier I guess.

Also the diff is hard to read because of the change to the indentation in `.travis.yml`, could you restore it?

File: src/Application/Cli/TravisAutoDeployCommand.php (hunk 1 above)
Comment (wysow) [inline src/Application/Cli/TravisAutoDeployCommand.php]: I updated the travis configuration to avoid this option, which were only there as a little hack to avoid re-doing the same action on each php version... So not needed anymore I think... But could require some documentation for sure...

File: src/Application/Cli/TravisAutoDeployCommand.php (hunk 2 above)
Comment (wysow) [inline src/Application/Cli/TravisAutoDeployCommand.php]: Sure I didn't think about this, I will update this PR ASAP

**Comment by wysow** [comment]
@mnapoli about the indentation of the `.travis.yml` file, it's due to the use of the travis cli tool to generate the `deploy` part, mainly needed to generate the encoded token, so maybe it could be great to leave this like this for the future as each time we will use it, the indentation will change... What do you think?

File: src/Application/Cli/TravisAutoDeployCommand.php (hunk 1 above)
Comment (mnapoli) [inline src/Application/Cli/TravisAutoDeployCommand.php]: Yes but users of Couscous might be using it, so even though we will not use it ourselves we must not break this

Comment (wysow) [inline src/Application/Cli/TravisAutoDeployCommand.php]: OK will revert this, but maybe it could be great to mark it as deprecated for next major...

**Comment by wysow** [comment]
@mnapoli any news on this one?

**Comment by mnapoli** [comment]
Sorry this got out of my todo list somehow

File: src/Application/Cli/TravisAutoDeployCommand.php [hunk 3]
```diff
@@ -101,19 +102,21 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
+            trigger_error('Deprecated option "php-version" called.', E_USER_NOTICE);
             $output->writeln('<comment>This version of the documentation is already deployed</comment>');
 
             return;
         }
 
+        // set git user data
+        $output->writeln('<info>Setting up git user</info>');
+        $this->commandRunner->run('git config --global user.name "${GIT_NAME}"');
+        $this->commandRunner->run('git config --global user.email "${GIT_EMAIL}"');
```
Comment (mnapoli) [inline src/Application/Cli/TravisAutoDeployCommand.php]: Why this change?

Comment (wysow) [inline src/Application/Cli/TravisAutoDeployCommand.php]: Because I thought that we should not change the git user unless we can really deploy the docs as per the `if` call just above...

**Comment by mnapoli** [comment]
Thanks!
